code/modules/lighting/lighting_atom.dm

```diff
@@ -63,7 +63,7 @@
 		if(!isnull(set_light_on(l_on)))
 			needs_update = TRUE
 
-	if(needs_update)
+	if(needs_update && light_system == STATIC_LIGHT)
```

Comment:
Don't do this, it will prevent the stack_trace from happening, which we want to see.